### ISSUE TITLE
“special characters” should be allowed for anchors.

### ISSUE THREAD

**Issue by tastytea** [asciidoctor/asciidoctor#3307]
``` shellsession
% asciidoctor --version
Asciidoctor 2.0.9 [https://asciidoctor.org]
Runtime Environment (ruby 2.4.5p335 (2018-10-18 revision 65137) [x86_64-linux]) (lc:UTF-8 fs:UTF-8 in:UTF-8 ex:UTF-8)
```

The following document does not convert to the expected HTML5 output.

``` asciidoc
= Test

anchor:CO₂[]abc
```

I used the command `asciidoctor --backend=html5 --out-file=out.html in.adoc`.

According to the [W3C Recommendation](https://www.w3.org/TR/html5/dom.html#element-attrdef-global-id), the `id` attribute has to conform only to these rules:

* Must contain at least one character.
* Must not contain any space characters.

> There are no other restrictions on what form an ID can take; in particular, IDs can consist of just digits, start with a digit, start with an underscore, consist of just punctuation, etc.

But the output is:

``` html
<div class="paragraph">
<p>anchor:CO₂[]abc</p>
</div>
```

I have similar results with `&`, `%`, `/`, `→` and anchors starting with a digit. Please allow “special characters” in anchors.

**Comment by cirosantilli** [comment]
Related: https://github.com/asciidoctor/asciidoctor/issues/1540

**Comment by mojavelinux** [comment]
> According to the W3C Recommendation, the id attribute has to conform only to these rules:

Yes, but HTML isn't the only output format for AsciiDoc. Using other characters would lead to invalid DocBook (and likely EPUB 3 too, which still conforms to XML rules in some versions and readers).

**Comment by cirosantilli** [comment]
One possible solution would be to let it work fine on HTML5 output, but blow up on other outputs.

**Comment by mojavelinux** [comment]
Right. Agreed. I just wanted to make sure to frame the conversation.
